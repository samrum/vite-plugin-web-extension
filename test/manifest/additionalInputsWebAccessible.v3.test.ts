import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("additionalInputsWebAccessible");

runManifestV3Test("additionalInputsWebAccessible", () => ({}), {
  additionalInputs: {
    scripts: [
      `${resourceDir}/script1.js`,
      {
        fileName: `${resourceDir}/script2.js`,
        webAccessible: false,
      },
      {
        fileName: `${resourceDir}/script3.js`,
        webAccessible: true,
      },
      {
        fileName: `${resourceDir}/script4.js`,
        webAccessible: {
          matches: ["https://example.com/"],
        },
      },
      {
        fileName: `${resourceDir}/script5.ts`,
        webAccessible: {
          matches: ["https://example.com/"],
          includeEntryFile: true,
        },
      },
      {
        fileName: `${resourceDir}/script6.js`,
        webAccessible: {
          extensionIds: ["oilkjaldkfjlasdf"],
        },
      },
      {
        fileName: `${resourceDir}/script7.js`,
        webAccessible: {
          extensionIds: ["oilkjaldkfjlasdf"],
          includeEntryFile: true,
        },
      },
      {
        fileName: `${resourceDir}/chunkedScript1.js`,
        webAccessible: {
          matches: ["https://example.com/"],
          includeEntryFile: true,
        },
      },
      {
        fileName: `${resourceDir}/chunkedScript2.js`,
        webAccessible: {
          matches: ["https://example.com/"],
        },
      },
    ],
  },
});
