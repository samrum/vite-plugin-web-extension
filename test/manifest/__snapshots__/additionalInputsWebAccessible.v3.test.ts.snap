// Vitest Snapshot v1

exports[`additionalInputsWebAccessible - Manifest V3 1`] = `
[
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script1.js",
    "name": undefined,
    "source": "console.log(\\"script1\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script2.js",
    "name": undefined,
    "source": "console.log(\\"script2\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script3.js",
    "name": undefined,
    "source": "console.log(\\"script3\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script4.js",
    "name": undefined,
    "source": "console.log(\\"script4\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script5.js",
    "name": undefined,
    "source": "console.log(\\"script5\\");
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/additionalInputsWebAccessible/script6.js",
    "name": undefined,
    "source": "console.log(\\"script6\\");
",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"web_accessible_resources\\": [
    {
      \\"resources\\": [
        \\"test/manifest/resources/additionalInputsWebAccessible/script3.js\\"
      ],
      \\"matches\\": [
        \\"<all_urls>\\"
      ]
    },
    {
      \\"resources\\": [
        \\"test/manifest/resources/additionalInputsWebAccessible/script4.js\\"
      ],
      \\"matches\\": [
        \\"https://example.com/\\"
      ]
    },
    {
      \\"resources\\": [
        \\"test/manifest/resources/additionalInputsWebAccessible/script5.js\\"
      ],
      \\"extension_ids\\": [
        \\"oilkjaldkfjlasdf\\"
      ]
    },
    {
      \\"resources\\": [
        \\"test/manifest/resources/additionalInputsWebAccessible/script6.js\\"
      ],
      \\"matches\\": [
        \\"https://example.com/\\"
      ],
      \\"use_dynamic_url\\": true
    }
  ]
}",
    "type": "asset",
  },
]
`;
